### PR TITLE
Fix Content-Type header checking of React/RCTJavascriptLoader.mm #15791

### PR DESCRIPTION
**Pull request by justinr1234**
To fix this issue: https://github.com/facebook/react-native/issues/15791

### CODE REVIEW

**Comment by justinr1234** [comment]
Not sure why the build failed, looks to be transient?

**Comment by justinr1234** [comment]
@shergin I can take a look, but can you elaborate on the motivation? It appears your suggestion would be equivalent except requiring additional temporary storage and processing?

**Comment by janicduplessis** [comment]
@shergin @justinr1234 We could also just use a regex, something like `/^(?:application|text)\/javascript/` should do the trick.

**Comment by justinr1234** [comment]
@janicduplessis @shergin I can make changes, I'm just confused what the purpose is and what's gained by changing from the current form. Thanks, it helps me understand going forward to learn the preferences of the project.

**Comment by ide** [comment]
The check in this PR doesn't look at the structure of a MIME type string. It just checks the prefix -- what if you have `text/javascript2`? We don't want to match that unintentionally.

**Comment by justinr1234** [comment]
@ide thanks, good point, I'll update it

**Comment by shergin** [comment]
BTW, I think using a regexp is overkill here.

**Comment by janicduplessis** [comment]
In that case @shergin suggestion to create a substring up to optional ; would work well.

File: React/Base/RCTJavaScriptLoader.mm [hunk 1]
```diff
@@ -274,9 +274,11 @@ static void attemptAsynchronousLoadOfBundleAtURL(NSURL *scriptURL, RCTSourceLoad
 
     // Validate that the packager actually returned javascript.
     NSString *contentType = headers[@"Content-Type"];
-    if (![contentType isEqualToString:@"application/javascript"] &&
-        ![contentType isEqualToString:@"text/javascript"]) {
-      NSString *description = [NSString stringWithFormat:@"Expected Content-Type to be 'application/javascript' or 'text/javascript', but got '%@'.", contentType];
+    NSArray<NSString *> *contentTypeArr = [contentType componentsSeparatedByString:@";"];
```
Comment (shergin) [inline React/Base/RCTJavaScriptLoader.mm]: Please don't use abbreviations.
`NSString *mimeType = [[contentType componentSeparatedByString:@";"] firstObject]` should be enough, right?

**Comment by janicduplessis** [comment]
@shergin Does this looks good now?

**Comment by justinr1234** [comment]
@shergin did this get merged somewhere?

**Comment by ide** [comment]
The commit above shows where it landed - https://github.com/facebook/react-native/commit/3ae45d5cb8b9aaca568227ca00cc304e053bf073

**Pull request closed without merging**